src/lib/ng-v5/entry-point/ts/compile-ngc.transform.ts

```diff
@@ -1,20 +1,35 @@
 import { Transform, transformFromPromise } from '../../../brocc/transform';
 import { compileSourceFiles } from '../../../ngc/compile-source-files';
+import { TsConfig } from '../../../ts/tsconfig';
 import * as log from '../../../util/log';
-import { isEntryPointInProgress, isTypeScriptSources, TypeScriptSourceNode } from '../../nodes';
+import { isEntryPointInProgress, isTypeScriptSources, TypeScriptSourceNode, isEntryPoint } from '../../nodes';
 
 export const compileNgcTransform: Transform = transformFromPromise(async graph => {
   log.info(`Compiling TypeScript sources through ngc`);
   const entryPoint = graph.find(isEntryPointInProgress());
   const tsSources: TypeScriptSourceNode = entryPoint.find(isTypeScriptSources) as TypeScriptSourceNode;
+  const tsConfig: TsConfig = entryPoint.data.tsConfig;
+
+  // Add paths mappings for dependencies
+  const entryPointDeps = entryPoint.filter(isEntryPoint);
+  if (entryPointDeps.length > 0) {
+    if (!tsConfig.options.paths) {
+      tsConfig.options.paths = {};
+    }
+
+    for (let dep of entryPointDeps) {
+      const depModuleId = dep.data.entryPoint.moduleId;
+
+      if (!tsConfig.options.paths[depModuleId]) {
+        tsConfig.options.paths[depModuleId] = [];
+      }
+      tsConfig.options.paths[depModuleId].push(dep.data.es2015EntryFile);
```

Comment:
Hi all,

After some debugging, I found out that Typescript doesn't like when you leave the ".js" extension file in its paths.

So something like this, will work to use a JS file directly but it won't let typescript find the matching declaration files :
```
"paths": {
  "@sample/secondary/shared": [
    ".ng_pkg_build/sample/out/shared.js"
  ]
}
```

Instead, we should get something like this :
```
"paths": {
  "@sample/secondary/shared": [
    ".ng_pkg_build/sample/out/shared"
  ]
}
```

This will allow Typescript to find the matching declaration files.

So, I wrote something like :
```
import * as path from 'path';
...
const depFilePath = path.resolve(
  path.dirname(dep.data.es2015EntryFile),
  path.basename(dep.data.es2015EntryFile, '.js')
);

tsConfig.options.paths[depModuleId].push(depFilePath);
```

By the way, everything else is working fine on my project.

Hope it will help